vpr/src/place/timing/PlacerCriticalities.h

```diff
@@ -0,0 +1,199 @@
+
+#pragma once
+
+#include "vtr_vec_id_set.h"
+#include "timing_info_fwd.h"
+#include "clustered_netlist_utils.h"
+#include "place_delay_model.h"
+#include "vpr_net_pins_matrix.h"
+
+/**
+ * @brief Saves the placement criticality parameters
+ *
+ * crit_exponent: The criticality exponent used to sharpen the criticalities
+ * crit_limit:    The limit to consider a pin as timing critical
+ */
+struct PlaceCritParams {
+    float crit_exponent;
+    float crit_limit;
+};
+
+/**
+ * @brief PlacerCriticalities returns the clustered netlist connection criticalities
+ *        used by the placer ('sharpened' by a criticality exponent).
+ *
+ * Usage
+ * =====
+ * This class also serves to map atom netlist level criticalites (i.e. on AtomPinIds)
+ * to the clustered netlist (i.e. ClusterPinIds) used during placement.
+ *
+ * Criticalities are updated by update_criticalities(), given that `update_enabled` is
+ * set to true. It will update criticalities based on the atom netlist connection
+ * criticalities provided by the passed in SetupTimingInfo.
+ *
+ * This process can be done incrementally, based on the modified connections/AtomPinIds
+ * returned by SetupTimingInfo. However, the set returned only reflects the connections
+ * changed by the last call to the timing info update.
+ *
+ * Therefore, if SetupTimingInfo is updated twice in succession without criticalities
```

Comment:
I don't find this paragraph easy to understand ... if you understand it, please take a crack at clarifying it.